pkgs/servers/mautrix-signal/default.nix

```diff
@@ -32,16 +35,30 @@ buildGoModule rec {
     })
   ];
 
-  buildInputs = (lib.optional (!withGoolm) olm) ++ [
-    # must match the version used in https://github.com/mautrix/signal/tree/main/pkg/libsignalgo
-    # see https://github.com/mautrix/signal/issues/401
-    libsignal-ffi
-  ];
+  buildInputs =
+    (lib.optional (!withGoolm) olm)
+    ++ (lib.optional withGoolm stdenv.cc.cc.lib)
+    ++ [
+      # must match the version used in https://github.com/mautrix/signal/tree/main/pkg/libsignalgo
+      # see https://github.com/mautrix/signal/issues/401
+      libsignal-ffi
+    ];
+
   tags = lib.optional withGoolm "goolm";
 
+  CGO_LDFLAGS = [ "-lstdc++" ];
```

Comment:
```suggestion
  CGO_LDFLAGS = lib.optional withGoolm "-lstdc++";
```

Reply:
This is also needed by the libolm-variant, as well as the goolm, at least with `doCheck = true` - otherwise, the tests fail to link to `libstdc++`.

Reply:
We don't want to unnecessarily link to stdc++ for the binaries produced with the libolm variant though, so maybe this should be moved to `preCheck` instead as `export CGO_LDFLAGS`.

Reply:
Yeah, right, makes sense. I'll see what I can come up with :+1: